qpc/insights/tests_insights_configure.py

```diff
@@ -0,0 +1,86 @@
+        sys.stderr = HushUpStderr()
+
+    def tearDown(self):
+        """Remove test case setup."""
+        # Restore stderr
+        sys.stderr = self.orig_stderr
+
+    def test_insights_config_bad_port(self):
+        """Testing insights configure when receiving bad port."""
+        with self.assertRaises(SystemExit):
+            sys.argv = ['/bin/qpc', 'insights', 'config',
+                        '--port', 'abc']
+            CLI().main()
+
+    def test_success_default_config_no_args(self):
+        """Testing if method returns default config dict when no arguments."""
+        config_out = StringIO()
+        sys.argv = ['/bin/qpc', 'insights', 'config']
+        with redirect_stdout(config_out):
+            CLI().main()
+        config = read_insights_config()
+        self.assertEqual(config['host'], DEFAULT_INSIGHTS_CONFIG['host'])
+        self.assertEqual(config['port'], DEFAULT_INSIGHTS_CONFIG['port'])
+        self.assertEqual(config['use_http'], DEFAULT_INSIGHTS_CONFIG['use_http'])
+
+    def test_success_config_insights(self):
+        """Testing insights configure green path."""
+        config_out = StringIO()
+        sys.argv = ['/bin/qpc', 'insights', 'config',
+                    '--host', 'console.insights.test', '--port', '200', '--use-http']
+        with redirect_stdout(config_out):
+            CLI().main()
+            config = read_insights_config()
+            self.assertEqual(config['host'], 'console.insights.test')
+            self.assertEqual(config['port'], 200)
+            self.assertEqual(config['use_http'], True)
+
+    def test_insights_config_default_host(self):
+        """Testing insights configure default host."""
+        sys.argv = ['/bin/qpc', 'insights', 'config',
+                    '--port', '200']
+        CLI().main()
+        config = read_insights_config()
+        self.assertEqual(config['host'], DEFAULT_INSIGHTS_CONFIG['host'])
+        self.assertEqual(config['port'], 200)
+        self.assertEqual(config['use_http'], DEFAULT_INSIGHTS_CONFIG['use_http'])
+
+    def test_insights_config_default_port(self):
+        """Testing insights configure default port."""
+        sys.argv = ['/bin/qpc', 'insights', 'config',
+                    '--host', 'console.insights.test']
+        CLI().main()
+        config = read_insights_config()
+        self.assertEqual(config['host'], 'console.insights.test')
+        self.assertEqual(config['port'], DEFAULT_INSIGHTS_CONFIG['port'])
+        self.assertEqual(config['use_http'], DEFAULT_INSIGHTS_CONFIG['use_http'])
+
+    def test_invalid_configuration(self):
+        """Test reading bad JSON on cli start."""
+        write_insights_config({})
+
+        config = read_insights_config()
+        self.assertEqual(config['host'], DEFAULT_INSIGHTS_CONFIG['host'])
+        self.assertEqual(config['port'], DEFAULT_INSIGHTS_CONFIG['port'])
+        self.assertEqual(config['use_http'], DEFAULT_INSIGHTS_CONFIG['use_http'])
```

Comment:
In this case you could use `self.assertDictEqual(config, DEFAULT_INSIGHTS_CONFIG)`

Reply:
good catch!